### ISSUE TITLE
[v.6.xx - Safari BUG ONLY] IconLayers not rendered if first layer is ScatterplotLayer (React Example with CodeSandbox)

### ISSUE THREAD

**Issue by bendersej** [uber/deck.gl#2171]
Hi there!

I wanted to report a bug that I ran into **only in Safari**, it's not critical as a workaround is to simply pass in a Layer different than `Scatterplot`. 

#### Actual Result
When the first layer passed to `DeckGL` is a `ScatterplotLayer`, the subsequent `IconLayers` are not rendered.

#### Expected Result
All layers should be rendered, similarly to how `Chrome` or `Firefox` render the layers.

#### Reproduce Steps
Pass in an array of layers with at least one `ScatterplotLayer` and one `IconLayer`, where the `ScatterplotLayer` is first: **[**`ScatterPlotLayer`, `IconLayer`**]**.

The map will only render the `ScatterplotLayer`.

#### Code Example
I created a mini project to help understand what's happening: https://codesandbox.io/s/8yw9nnl270.
- Open in `Chrome` or `Firefox` to see the expected result. 
- Open in `Safari` to see the bug: the `Icon` is not rendered.

Keep up the good work!

**Comment by ibgreen** [comment]
@bendersej Thanks for reporting. If this renders fine on Chrome this is most likely related to Safari's handling of "constant attribute 0".

There is a bug open on luma.gl v6 to address this. Unfortunately, it can't be patched quickly but a fix is in progress.

FWIW, at the moment, luma.gl v5.3 does better with Safari.

**Comment by bendersej** [comment]
@ibgreen, thanks for the prompt answer! No worries :)

I didn't know about the open issue in `luma.gl`. I guess this issue can then serve as a reference in case someone else run into this issue.

**Comment by dekkofilms** [comment]
I think I'm having a similar issue, but it seems to be more sporadic and happening with only the `TextLayer`. Some of the `TextLayer`s appear and some don't and the order of layers being composed on top of each other are always the same.

**Comment by dekkofilms** [comment]
Also, is there a potential timeline on when that fix would be out?

**Comment by ibgreen** [comment]
> Also, is this an accurate timeline of when it will be out?

We'll try. It will for sure be solved in v6.1 before the end of this month, and trying to get something ready before that. It is not a small fix, as can be seen in WIP PR. And v5.3 is an option until then.

**Comment by dekkofilms** [comment]
Also @ibgreen, I just realized I am having this issue with luma.gl version being locked at v5.3.0.

```
└─┬ deck.gl@5.3.0
  └─┬ @deck.gl/core@5.3.0
    └── luma.gl@5.3.0
```

**Comment by ibgreen** [comment]
> I just realized I am having this issue with luma.gl version being locked at v5.3.0.


@dekkofilms Yes, it could happen. 

In fact, deck.gl 5.3 is not immune to the problem, although it uses the problematic "constant attributes" to a much lesser degree than 6.0 does.

We have just published 6.1-beta.1 of deck and luma which contains the long-term fix for this issue, we are testing it now. 

If you give it a spin, let us know!

**Comment by dekkofilms** [comment]
Awesome, I will try to give the new version a try this week, and for sure let you know if our issue has been resolved. 👍 Thanks so much, @ibgreen
